src/service.ts

```diff
@@ -55,175 +62,177 @@ interface ServiceOptions {
   token?: string;
 }
 
-/**
- * Service is a base class, meant to be inherited from by a "service," like
- * BigQuery or Storage.
- *
- * This handles making authenticated requests by exposing a `makeReq_` function.
- *
- * @constructor
- * @alias module:common/service
- *
- * @param {object} config - Configuration object.
- * @param {string} config.baseUrl - The base URL to make API requests to.
- * @param {string[]} config.scopes - The scopes required for the request.
- * @param {object=} options - [Configuration object](#/docs).
- */
-function Service(config: ServiceConfig, options?: ServiceOptions) {
-  options = options || {};
-
-  util.privatize(this, 'baseUrl', config.baseUrl);
-  util.privatize(this, 'globalInterceptors', arrify(options.interceptors_));
-  util.privatize(this, 'interceptors', []);
-  util.privatize(this, 'packageJson', config.packageJson);
-  util.privatize(this, 'projectId', options.projectId || PROJECT_ID_TOKEN);
-  util.privatize(this, 'projectIdRequired', config.projectIdRequired !== false);
-  util.privatize(this, 'Promise', options.promise || Promise);
-
-  const reqCfg = extend({}, config, {
-    projectIdRequired: this.projectIdRequired,
-    projectId: this.projectId,
-    credentials: options.credentials,
-    keyFile: options.keyFilename,
-    email: options.email,
-    token: options.token,
-  });
-
-  util.privatize(
-    this,
-    'makeAuthenticatedRequest',
-    util.makeAuthenticatedRequestFactory(reqCfg)
-  );
-  util.privatize(this, 'authClient', this.makeAuthenticatedRequest.authClient);
-  util.privatize(
-    this,
-    'getCredentials',
-    this.makeAuthenticatedRequest.getCredentials
-  );
-
-  const isCloudFunctionEnv = !!process.env.FUNCTION_NAME;
-
-  if (isCloudFunctionEnv) {
-    this.interceptors.push({
-      request(reqOpts: r.Options) {
-        reqOpts.forever = false;
-        return reqOpts;
-      },
+export class Service {
+
+  private baseUrl: string;
+  private globalInterceptors;
+  private interceptors: ({ request(opts: r.Options): r.Options})[];
+  private packageJson: PackageJson;
+  private projectId: string;
+  private projectIdRequired: boolean;
+  private Promise: Promise<{}>;
+  // TODO: make this private
+  makeAuthenticatedRequest;
```

Comment:
Can you add types to these objects? Looking at this file itself I'm not sure if it can potentially return a `Promise` or not (I sometimes see `return this.makeAuthenticatedRequest(opts, callback)` which seems strange to me)